tests/ui/lint/unused/unused-associated-item.rs

```diff
@@ -0,0 +1,21 @@
+//@ check-pass
```

Comment:
This is a user-facing consequence of this change and how it interacts w/ associated type bounds.

Previously the `must_use` implementation used `item_bounds`, which means we see the `Future` trait in the associated item bound of `impl Factory<Output: Future>` even if it's not bounding the RPIT itself.